test/unit/property-list.test.js

```diff
@@ -720,6 +720,162 @@ describe('PropertyList', function () {
             // should have removed the array of values
             expect(list.reference).to.not.have.property('key1');
         });
+
+        it('should correctly remove an element by direct reference', function () {
+            var entity = new FakeType({ keyAttr: 'key', valeu: 'val' }),
+                list = new PropertyList(FakeType, {}, [entity]);
+
+            list.remove(entity);
+
+            // should have removed the correct element from the reference map, and removed the array of values
+            expect(list.reference).to.not.have.property('key');
+        });
+    });
+
+    describe('.each', function () {
+        var FakeType = function (options) {
+            this.key = options.key;
+            this.value = options.value;
+        };
+
+        it('should handle non-function iterators correctly', function () {
```

Comment:
The difference comes from the implementation of `_.forEach`, which sanitizes the iterator.